File: app/src/main/java/com/linheimx/zimudog/vp/search/IContract.java
package com.linheimx.zimudog.vp.search;

import android.support.annotation.NonNull;

import com.linheimx.lspider.god.GodItem;
import com.linheimx.lspider.god.IMovie;
import com.linheimx.zimudog.vp.base.IPresenter;

import java.util.List;

/**
 * Created by x1c on 2017/5/1.
 */

public interface IContract {

    interface P extends IPresenter {
        void searchByKW(@NonNull String keyWord);

        void loadMore();

        void cancelSearch();

        void release();
    }

    interface V {
        void showItems(List<GodItem> movies, boolean hasMore);

        void showLoadingError();
    }
}

File: app/src/main/java/com/linheimx/zimudog/vp/search/IContract.kt
package com.linheimx.zimudog.vp.search

/**
 * Created by x1c on 2017/5/1.
 */

interface IContract {

    interface P {
        fun searchByKW(keyWord: String)

        fun loadMore()

        fun cancelSearch()

        fun release()
    }

    interface V {
        fun showItems(movies: List<com.linheimx.zimudog.m.bean.Movie>, hasMore: Boolean)

        fun showLoadingError(ex: Throwable)

        fun showNoDataView()
    }
}
